pkg/client/query/client_test.go

```diff
@@ -0,0 +1,55 @@
+//go:build integration
+
+package query_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"pocket/internal/testclient/testquery"
+)
+
+const (
+	msgReceiveTimeout = 5 * time.Second
+	msgReceiveLimit   = 2
+)
+
+func TestQueryClient_SubscribeWithQuery_NewBlock(t *testing.T) {
```

Comment:
For my own knowledge does this test require the chain to be running, I can't see how this wouldn't always timeout without the chain running

Reply:
This test does not integrate a real server. It mocks the connection's `Receive` function to produce values for the observable's producer to receive.